### DISCUSSION TITLE
Lingering Appium Process (NodeJS) After Testing Has Finished

### DISCUSSION THREAD

**Discussion by samuelfreiberg** [webdriverio/webdriverio#11225] [General]
Hi All,

This has been a problem for quite some time now. I'm using WDIO v7.30.1 test runner with Appium on Windows.

After executing WDIO test runner and running all tests (testing has completely finished), an Appium process still continues to run. I found this out because, if I try to execute tests back to back (ie. execute wdio test runner script (successfully). Re-run in same command prompt), I get an error stating:
```
22T16:06:27.603Z ERROR @wdio/cli:utils: A service failed in the 'onPrepare' hook
Error: Appium exited before timeout (exit code: 1)
    at ChildProcess.<anonymous> (C:\fabric-internal\common\temp\node_modules\.pnpm\@wdio+appium-service@7.30.0_ihgzpspkhfrpgr2ovt6c6ys5kq\node_modules\@wdio\appium-service\build\launcher.js:110:22)
    at Object.onceWrapper (events.js:520:26)
    at ChildProcess.emit (events.js:400:28)
    at ChildProcess.emit (domain.js:475:12)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:285:12

```

After digging, I found that this is because an Appium process is still running, and it's causing this error.

If I go in my task manager and kill the process, I'm able to run successfully again. 

Does anyone have any idea why this is happening? Why would there be a lingering process after test execution?

"webdriverio": "7.30.1"
 "@wdio/appium-service": "7.30.0",
 "@wdio/cli": "7.30.1",
 "@wdio/jasmine-framework": "7.26.0",
 "@wdio/local-runner": "7.30.1",
  "appium": "2.0.0",

**Comment by christian-bromann** [comment]
> Does anyone have any idea why this is happening? Why would there be a lingering process after test execution?

WebdriverIO is suppose to shut down the Appium process if the session ends properly (no ctrl + c or other manual process interruptions). Maybe you can debug the Appium service code in your node modules e.e. `/node_modules/@wdio/appium-service/build/launcher.js` and see if the `onComplete` hook is called properly.

**Reply by samuelfreiberg** [reply]
Thanks for this. Turns out it is going through that code correctly.

I found that the extra NodeJS process is spawned from Appium's index file (source). I'm not entirely sure why it's continuing to stay alive. I'd be curious if anyone else is testing on Win32 if they're having this issue.

**Reply by mattfldev** [reply]
I am seeing this as well on Win32.  I seem to have to run twice otherwise I get that error
